app/Http/Controllers/UsersController.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Http\Requests;
+use Prettus\Validator\Contracts\ValidatorInterface;
+use Prettus\Validator\Exceptions\ValidatorException;
+use App\Http\Requests\UserCreateRequest;
+use App\Http\Requests\UserUpdateRequest;
+use App\Repositories\UserRepository;
+use App\Validators\UserValidator;
+use Hash;
+
+/**
+ * Class UsersController.
+ *
+ * @package namespace App\Http\Controllers;
+ */
+class UsersController extends Controller
+{
+    /**
+     * @var UserRepository
+     */
+    protected $repository;
+
+    /**
+     * @var UserValidator
+     */
+    protected $validator;
+
+    /**
+     * UsersController constructor.
+     *
+     * @param UserRepository $repository
+     * @param UserValidator $validator
+     */
+    public function __construct(UserRepository $repository, UserValidator $validator)
+    {
+        $this->repository = $repository;
+        $this->validator  = $validator;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $this->repository->pushCriteria(app('Prettus\Repository\Criteria\RequestCriteria'));
+        $users = $this->repository->all();
+
+        if (request()->wantsJson()) {
+
+            return response()->json([
+                'data' => $users,
+            ]);
+        }
+
+        return view('users.index', compact('users'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  UserCreateRequest $request
+     *
+     * @return \Illuminate\Http\Response
+     *
+     * @throws \Prettus\Validator\Exceptions\ValidatorException
+     */
+    public function store(UserCreateRequest $request)
+    {
+        try {
+
+            $newRequest = $request->all();
+            $newRequest['password'] = Hash::make($request->password);
```

Comment:
line#78 try to hash password by using Laravel Mutators

Reply:
```
public function setPasswordAttribute($value)
{
    $this->attributes['password'] = Hash::make($value)
}
```
I think this one won't work because when you login with password it will mutate.